frontend/packages/console-dynamic-plugin-sdk/src/extensions/pages.ts

```diff
@@ -14,7 +33,65 @@ export type StandaloneRoutePage = ExtensionDeclaration<
   }
 >;
 
+/** Adds new page to Console router. */
+export type RoutePage = ExtensionDeclaration<
+  'console.page/route',
+  {
+    /** The React component to load for the page. */
+    component: CodeRef<React.Component<RouteComponentProps>>;
```

Comment:
`React.Component` refers to class components only. `React.FC` refers to function components only.

Please use `React.ComponentType` here, as well as in `StandaloneRoutePage` extension declaration above.

```ts
type ComponentType<P = {}> = ComponentClass<P> | FunctionComponent<P>;
```

In general, all extension declarations should use `React.ComponentType` when referring to a React component.